streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -226,7 +257,7 @@ public void writeAsText(String filePath, String streamName, Serde<K> keySerde, S
         try {
 
             PrintStream printStream = new PrintStream(new FileOutputStream(filePath));
-            topology.addProcessor(name, new KeyValuePrinter<>(printStream, keySerde, valSerde, streamName), this.name);
+            topology.addProcessor(name, new KStreamPeek<>(printAction(printStream, keySerde, valSerde, streamName, null)), this.name);
```

Comment:
sorry  - please ignore my previous comment. I was ignoring the fact that the mapper returns a result.